gunicorn/arbiter.py

```diff
@@ -473,6 +473,8 @@ def manage_workers(self):
         Maintain the number of workers by spawning or killing
         as required.
         """
+        orig_num_workers = self.num_workers
+
         if len(self.WORKERS.keys()) < self.num_workers:
```

Comment:
In my experience, using a mutable `has_changed`-type flag tends to open up a class of errors where the flag does not get set appropriately after modification. Particularly where that modification happens in a separate method call (Can `kill_worker` modify `num_workers`? I assume so. Does it always? I don't know). That generally crops up after future code changes/maintenance.

It also makes the if-statement harder to reason about: does this flag represent that the value being printed out changed? That it _could_ have changed? _Should_ have? If I see multiple "3 workers" messages in sequence, is that expected or a bug?

This is a short method, but those are the reasons I tend to prefer the state-comparison approach when it's not a computationally difficult one.